### PR TITLE
Ensure S3 signed manifest URLs don't expire prematurely (#5503)

### LINKED ISSUE #5503: Ensure S3 signed manifest URLs don't expire prematurely
https://github.com/DataBiosphere/azul-private/issues/96#issuecomment-1707401453
>@hannes-ucsc: "Additionally, we should make sure that we sign S3 object URLs using fresh STS credentials so that they don't expire prematurely in case cached STS credentials are being used."

- [x] Security design review completed; the Resolution of this issue does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [x] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below

### PR DESCRIPTION
**Pull request by dsotirho-ucsc**
Connected issues: #5503

### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] On ZenHub, PR is connected to all issues it (partially) resolves
- [x] PR description links to connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] PR title references all connected issues
- [x] For each connected issue, there is at least one commit whose title references that issue

### Author (partiality)

- [x] Added `p` tag to titles of partial commits
- [x] Added `partial` label to PR <sub>or this PR completely resolves all connected issues</sub>
- [x] All connected issues are resolved partially <sub>or this PR does not have the `partial` label</sub>

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title

### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR <sub>or this PR does not require reindexing</sub>
- [x] PR and connected issue are labeled `API` <sub>or this PR does not modify a REST API</sub>
- [x] Added `a` (`A`) tag to commit title for backwards (in)compatible changes <sub>or this PR does not modify a REST API</sub>
- [x] Updated REST API version number in `app.py` <sub>or this PR does not modify a REST API</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `base` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Added `chained` label to this PR <sub>or this PR is not chained to another PR</sub>

### Author (upgrading deployments)

- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading deployments</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading deployments</sub>
- [x] Added `upgrade` label to PR <sub>or this PR does not require upgrading deployments</sub>

### Author (operator tasks)

- [x] Added checklist items for additional operator tasks <sub>or this PR does not require additional tasks</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the `prod` branch has no temporary hotfixes for any connected issues</sub>

### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR <sub>or this PR does not touch requirements*.txt</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

### Peer reviewer (after requesting changes)

Uncheck the *Author (before every review)* checklists.

### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] Requested review from system administrator
- [x] PR is assigned to system administrator

### System administrator (after requesting changes)

Uncheck the *before every review* checklists. Update the `N reviews` label.

### System administrator (after approval)

- [ ] Actually approved the PR
- [ ] Labeled connected issues as `demo` or `no demo`
- [ ] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [ ] Decided if PR can be labeled `no sandbox`
- [ ] PR title is appropriate as title of merge commit
- [ ] `N reviews` label is accurate
- [ ] Moved ticket to *Approved* column
- [ ] PR is assigned to current operator

### Operator (before pushing merge the commit)

- [ ] Checked `reindex` label and `r` commit title tag
- [ ] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [ ] PR has checklist items for upgrading instructions <sub>or PR is not labeled `upgrade`</sub>
- [ ] Squashed PR branch and rebased onto `develop`
- [ ] Sanity-checked history
- [ ] Pushed PR branch to GitHub
- [ ] Added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [ ] Pushed PR branch to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Pushed PR branch to GitLab `anvilprod` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passes in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Reviewed build logs for anomalies in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `dev`</sub>
- [ ] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvildev`</sub>
- [ ] Deleted unreferenced indices in `hammerbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvilprod`</sub>
- [ ] Started reindex in `sandbox` <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `anvilbox` <sub>or this PR does not require reindexing `anvildev`</sub>
- [ ] Started reindex in `hammerbox` <sub>or this PR does not require reindexing `anvilprod`</sub>
- [ ] Checked for failures in `sandbox` <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `anvilbox` <sub>or this PR does not require reindexing `anvildev`</sub>
- [ ] Checked for failures in `hammerbox` <sub>or this PR does not require reindexing `anvilprod`</sub>
- [ ] Title of merge commit starts with title from this PR
- [ ] Added PR reference to merge commit title
- [ ] Collected commit title tags in merge commit title <sub>but only include `p` if the PR is labeled `partial`</sub>
- [ ] Moved connected issues to Merged column in ZenHub
- [ ] Pushed merge commit to GitHub

### Operator (chain shortening)

- [ ] Changed the target branch of the blocked PR to `develop` <sub>or this PR is not labeled `base`</sub>
- [ ] Removed the `chained` label from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [ ] Removed the blocking relationship from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [ ] Removed the `base` label from this PR <sub>or this PR is not labeled `base`</sub>

### Operator (after pushing the merge commit)

- [ ] Deployed `dev.gitlab`
- [ ] Pushed merge commit to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Deployed `anvildev.gitlab`
- [ ] Pushed merge commit to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Deployed `anvilprod.gitlab`
- [ ] Pushed merge commit to GitLab `anvilprod` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passes on GitLab `dev`<sup>1</sup>
- [ ] Reviewed build logs for anomalies on GitLab `dev`<sup>1</sup>
- [ ] Build passes on GitLab `anvildev`<sup>1</sup>
- [ ] Reviewed build logs for anomalies on GitLab `anvildev`<sup>1</sup>
- [ ] Build passes on GitLab `anvilprod`<sup>1</sup>
- [ ] Reviewed build logs for anomalies on GitLab `anvilprod`<sup>1</sup>
- [ ] Deleted PR branch from GitHub
- [ ] Deleted PR branch from GitLab `dev`
- [ ] Deleted PR branch from GitLab `anvildev`
- [ ] Deleted PR branch from GitLab `anvilprod`

<sup>1</sup> When pushing the merge commit is skipped due to the PR being
labelled `no sandbox`, the next build triggered by a PR whose merge commit *is*
pushed determines this checklist item.

### Operator (reindex)

- [ ] Deleted unreferenced indices in `dev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `dev`</sub>
- [ ] Deleted unreferenced indices in `anvildev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvildev`</sub>
- [ ] Deleted unreferenced indices in `anvilprod` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvilprod`</sub>
- [ ] Considered deindexing individual sources in `dev` <sub>or this PR does not remove individual sources from existing catalogs in `dev`</sub>
- [ ] Considered deindexing individual sources in `anvildev` <sub>or this PR does not remove individual sources from existing catalogs in `anvildev`</sub>
- [ ] Considered deindexing individual sources in `anvilprod` <sub>or this PR does not remove individual sources from existing catalogs in `anvilprod`</sub>
- [ ] Considered indexing individual sources in `dev` <sub>or this PR does not merely add individual sources to existing catalogs in `dev`</sub>
- [ ] Considered indexing individual sources in `anvildev` <sub>or this PR does not merely add individual sources to existing catalogs in `anvildev`</sub>
- [ ] Considered indexing individual sources in `anvilprod` <sub>or this PR does not merely add individual sources to existing catalogs in `anvilprod`</sub>
- [ ] Started reindex in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [ ] Started reindex in `anvilprod` <sub>or this PR does not require reindexing `anvilprod`</sub>
- [ ] Checked for and triaged indexing failures in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for and triaged indexing failures in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [ ] Checked for and triaged indexing failures in `anvilprod` <sub>or this PR does not require reindexing `anvilprod`</sub>
- [ ] Emptied fail queues in `dev` deployment <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Emptied fail queues in `anvildev` deployment <sub>or this PR does not require reindexing `anvildev`</sub>
- [ ] Emptied fail queues in `anvilprod` deployment <sub>or this PR does not require reindexing `anvilprod`</sub>

### Operator

- [ ] Added CL item to `prod` promotion PR:
> - [ ] Deployed `prod.gitlab`
- [ ] PR is assigned to no one

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Comment by dsotirho-ucsc** [comment]
Tested generating a manifest on personal deployment.
Note: Identity `azul-service-daniel-manifest` assuming role of `azul-service-daniel`

[CloudWatch logs](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2024-02-09T22*3a08*3a07.725Z~start~'2024-02-09T22*3a07*3a05.402Z~timeType~'ABSOLUTE~tz~'LOCAL~editorString~'fields*20*40timestamp*2c*20*40log*2c*20*40message*0a*23*20*7c*20filter*20*40message*20like*20*2f73.222.193.39*7cassume*20role*2f*0a*7c*20filter*20*40message*20like*20*2fReceived*7cReturning*7cSTART*7cEND*7cassume*20role*2f*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'e74087176a2b98e9-b4fb9844-491194-de29b899-d9edac30ad316bd09d647a33~source~(~'*2faws*2flambda*2fazul-service-daniel-manifest~'*2faws*2flambda*2fazul-service-daniel))):
```
[
    {
        "@timestamp": "2024-02-09 22:07:14.832",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "START RequestId: 69f57e67-958f-4cc5-9576-a576dd2f36aa Version: $LATEST\n"
    },
    {
        "@timestamp": "2024-02-09 22:07:14.833",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "[INFO]\t2024-02-09T22:07:14.833Z\t69f57e67-958f-4cc5-9576-a576dd2f36aa\tazul.chalice\tReceived PUT request for '/fetch/manifest/files', with {\"query\": {\"catalog\": \"dcp3\", \"filters\": \"{\\n  \\\"cellCount\\\": {\\n    \\\"within\\\": [\\n      [\\n        10000,\\n        100000003300\\n      ]\\n    ]\\n  }\\n}\", \"format\": \"curl\"}, \"headers\": {\"accept\": \"*/*\", \"cloudfront-forwarded-proto\": \"https\", \"cloudfront-is-desktop-viewer\": \"true\", \"cloudfront-is-mobile-viewer\": \"false\", \"cloudfront-is-smarttv-viewer\": \"false\", \"cloudfront-is-tablet-viewer\": \"false\", \"cloudfront-viewer-asn\": \"7922\", \"cloudfront-viewer-country\": \"US\", \"host\": \"service.daniel.dev.singlecell.gi.ucsc.edu\", \"user-agent\": \"curl/8.4.0\", \"via\": \"2.0 969054bdb3a3ab0ab0c42213e2a06ed8.cloudfront.net (CloudFront)\", \"x-amz-cf-id\": \"8loTpvdYcxFWdRpMkGwxohyFSaPW3O5qpZTviYu3Azxp3awySqAZiA==\", \"x-amzn-trace-id\": \"Root=1-65c6a212-0b45fb7570e7d13f55f08971\", \"x-forwarded-for\": \"73.222.193.39, 70.132.61.161\", \"x-forwarded-port\": \"443\", \"x-forwarded-proto\": \"https\"}}."
    },
    {
        "@timestamp": "2024-02-09 22:07:15.229",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "[DEBUG]\t2024-02-09T22:07:15.229Z\t69f57e67-958f-4cc5-9576-a576dd2f36aa\tazul.chalice\tReturning 200 response with headers {\"Access-Control-Allow-Origin\": \"*\", \"Access-Control-Allow-Headers\": \"Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key\", \"Strict-Transport-Security\": \"max-age=31536000; includeSubDomains\"}. See next line for the first 1024 characters of the body.\n{\"Status\": 301, \"Location\": \"https://service.daniel.dev.singlecell.gi.ucsc.edu/fetch/manifest/files/k8QgQRdSuLIHZgnwmw1q7E-X09NFU9L_o8Bi90u14HBefacAAQ==\", \"Retry-After\": 1}"
    },
    {
        "@timestamp": "2024-02-09 22:07:15.230",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "END RequestId: 69f57e67-958f-4cc5-9576-a576dd2f36aa\n"
    },
    {
        "@timestamp": "2024-02-09 22:07:15.366",
        "@log": "122796619775:/aws/lambda/azul-service-daniel-manifest",
        "@message": "START RequestId: f883e93e-486a-4705-b264-a0440d238fb0 Version: $LATEST\n"
    },
    {
        "@timestamp": "2024-02-09 22:07:16.456",
        "@log": "122796619775:/aws/lambda/azul-service-daniel-manifest",
        "@message": "[INFO]\t2024-02-09T22:07:16.456Z\tf883e93e-486a-4705-b264-a0440d238fb0\tazul.deployment\tIdentity AROARZFZ7W77UDPWDWS7Y:azul-service-daniel-manifest with ARN arn:aws:sts::122796619775:assumed-role/azul-service-daniel/azul-service-daniel-manifest is about to assume role arn:aws:iam::122796619775:role/azul-service-daniel using session name AROARZFZ7W77UDPWDWS7Y.azul-service-daniel-manifest."
    },
    {
        "@timestamp": "2024-02-09 22:07:16.547",
        "@log": "122796619775:/aws/lambda/azul-service-daniel-manifest",
        "@message": "END RequestId: f883e93e-486a-4705-b264-a0440d238fb0\n"
    },
    {
        "@timestamp": "2024-02-09 22:07:27.970",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "START RequestId: 07062f47-6d38-42f9-bf85-6f2a1d65c997 Version: $LATEST\n"
    },
    {
        "@timestamp": "2024-02-09 22:07:27.971",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "[INFO]\t2024-02-09T22:07:27.971Z\t07062f47-6d38-42f9-bf85-6f2a1d65c997\tazul.chalice\tReceived GET request for '/fetch/manifest/files/k8QgQRdSuLIHZgnwmw1q7E-X09NFU9L_o8Bi90u14HBefacAAQ==', with {\"query\": null, \"headers\": {\"accept\": \"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7\", \"accept-encoding\": \"gzip, deflate, br\", \"accept-language\": \"en-US,en;q=0.9\", \"cloudfront-forwarded-proto\": \"https\", \"cloudfront-is-desktop-viewer\": \"true\", \"cloudfront-is-mobile-viewer\": \"false\", \"cloudfront-is-smarttv-viewer\": \"false\", \"cloudfront-is-tablet-viewer\": \"false\", \"cloudfront-viewer-asn\": \"7922\", \"cloudfront-viewer-country\": \"US\", \"cookie\": \"G_ENABLED_IDPS=google; cs9-prd-web-8-1810-PORTAL-PSJSESSIONID=gjCUbg0fXxUA-dMJybbzUQoSxc9Xrogq!1567185761; experimentation_subject_id=IjA2OTQ1NGZmLTUzYmQtNGFlMi05ZTA1LTRiY2RkZTg1ZGFmYiI%3D--98287bfbd2c1c6ba94965fee0bb034b6d1f8677e; hguid=1093029397_YWSQPQ01F29qhTjG1BBAlcHqKUtd; lcsrftoken=exfsf+JnH1Vkp/DKx6PJPO8ZCV6/Ntwr4h1w8RY+y7w=\", \"host\": \"service.daniel.dev.singlecell.gi.ucsc.edu\", \"sec-ch-ua\": \"\\\"Not A(Brand\\\";v=\\\"99\\\", \\\"Google Chrome\\\";v=\\\"121\\\", \\\"Chromium\\\";v=\\\"121\\\"\", \"sec-ch-ua-mobile\": \"?0\", \"sec-ch-ua-platform\": \"\\\"macOS\\\"\", \"sec-fetch-dest\": \"document\", \"sec-fetch-mode\": \"navigate\", \"sec-fetch-site\": \"none\", \"sec-fetch-user\": \"?1\", \"sec-gpc\": \"1\", \"upgrade-insecure-requests\": \"1\", \"user-agent\": \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36\", \"via\": \"2.0 6ecf574c848f26fb93b4bc63ae69709a.cloudfront.net (CloudFront)\", \"x-amz-cf-id\": \"dHPpz1D8HURLy8WCq9fZUtQhuaE6AkWc9sX7dSbDkHMF42j8aJjD3A==\", \"x-amzn-trace-id\": \"Root=1-65c6a21f-40fa354f5a5927a56286299c\", \"x-forwarded-for\": \"73.222.193.39, 70.132.61.90\", \"x-forwarded-port\": \"443\", \"x-forwarded-proto\": \"https\"}}."
    },
    {
        "@timestamp": "2024-02-09 22:07:28.056",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "[DEBUG]\t2024-02-09T22:07:28.056Z\t07062f47-6d38-42f9-bf85-6f2a1d65c997\tazul.chalice\tReturning 200 response with headers {\"Access-Control-Allow-Origin\": \"*\", \"Access-Control-Allow-Headers\": \"Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key\", \"Strict-Transport-Security\": \"max-age=31536000; includeSubDomains\"}. See next line for the first 1024 characters of the body.\n{\"Status\": 302, \"Location\": \"https://service.daniel.dev.singlecell.gi.ucsc.edu/manifest/files/ksQvlKRkY3AzpGN1cmzEEMK9jG-4Z1R3rCiMW7p1WWLEEK0UsyveoFVet0WOvluwlY3EIPGHxuZb07O05Le3l-rT_HuGA1mgAz8hiRFko2_R9X74\", \"CommandLine\": {\"cmd.exe\": \"curl.exe --location --fail \\\"https://service.daniel.dev.singlecell.gi.ucsc.edu/manifest/files/ksQvlKRkY3AzpGN1cmzEEMK9jG-4Z1R3rCiMW7p1WWLEEK0UsyveoFVet0WOvluwlY3EIPGHxuZb07O05Le3l-rT_HuGA1mgAz8hiRFko2_R9X74\\\" | curl.exe --fail-early --continue-at - --retry 2 --retry-delay 10 --config -\", \"bash\": \"curl --location --fail https://service.daniel.dev.singlecell.gi.ucsc.edu/manifest/files/ksQvlKRkY3AzpGN1cmzEEMK9jG-4Z1R3rCiMW7p1WWLEEK0UsyveoFVet0WOvluwlY3EIPGHxuZb07O05Le3l-rT_HuGA1mgAz8hiRFko2_R9X74 | curl --fail-early --continue-at - --retry 2 --retry-delay 10 --config -\"}}"
    },
    {
        "@timestamp": "2024-02-09 22:07:28.058",
        "@log": "122796619775:/aws/lambda/azul-service-daniel",
        "@message": "END RequestId: 07062f47-6d38-42f9-bf85-6f2a1d65c997\n"
    }
]
```

**Comment by dsotirho-ucsc** [comment]
[5503_IT_2024-02-09.txt](https://github.com/DataBiosphere/azul/files/14226935/5503_IT_2024-02-09.txt)

File: src/azul/service/manifest_service.py [hunk 1]
```diff
@@ -705,7 +705,9 @@ def _presign_manifest(self,
         if not generator_cls.use_content_disposition_file_name:
             file_name = None
         object_key = generator_cls.s3_object_key(manifest_key)
-        presigned_url = self.storage_service.get_presigned_url(object_key, file_name)
+        role_arn = config.lambda_role_arn('service')
+        with aws.assumed_role_credentials(role_arn=role_arn):
```
Comment (achave11-ucsc) [inline src/azul/service/manifest_service.py]: My personal preference would be to inline this section, but I leave it at your discretion.
```suggestion
        with aws.assumed_role_credentials(config.lambda_role_arn('service')):
```

Do consider that the argument to this method is conventionally treated as a positional, referencing https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#125method-and-function-arguments the first bulleted item in the section permits the omission of the `role_arn` keyword, however you may want to consider enforcing the last bulleted item.

File: terraform/gitlab/gitlab.tf.json.template.py [hunk 2]
```diff
@@ -673,6 +673,20 @@ def qq(*words):
                         'resources': [
                             '*'
                         ]
+                    },
+
+                    # Extend boundary to allow the service Lambda role to assume
+                    # itself
```
Comment (achave11-ucsc) [inline terraform/gitlab/gitlab.tf.json.template.py]: Consider dropping the fluff "Extend boundary to" and enhancing the info comment with a one-liner of the reason why the service Lambda role must assume itself. The modifications to this section inform of the boundary extension.

**Review by achave11-ucsc — CHANGES_REQUESTED** [review]
Small nits, otherwise good work!

**Review by achave11-ucsc — DISMISSED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
I'd like you to follow the demo instructions for your personal deployment, and post the terminal transcript, before I review this.

https://github.com/DataBiosphere/azul/issues/5503#issuecomment-1942301120

**Comment by dsotirho-ucsc** [comment]
Closing abandoned PR. Attached issue was closed as [invalid](https://github.com/DataBiosphere/azul/issues/5503#issuecomment-1962005938).

**Pull request closed without merging**